sparkle/configurator/configurator.py

```diff
@@ -4,152 +4,82 @@
+from sparkle.types.objective import PerformanceMeasure
 
 
 class Configurator:
-    """Generic class to use different configurators like SMAC."""
+    """Abstact class to use different configurators like SMAC."""
+    configurator_cli_path = Path("sparkle/configurator/configurator_cli.py")
 
-    def __init__(self: Configurator, configurator_path: Path, executable_path: Path,
-                 settings_path: Path, result_path: Path, configurator_target: Path,
+    def __init__(self: Configurator, validator: Validator, output_path: Path,
+                 executable_path: Path, settings_path: Path, configurator_target: Path,
                  tmp_path: Path = None, multi_objective_support: bool = False) -> None:
         """Initialize Configurator.
 
         Args:
-            configurator_path: Path to the configurator directory
+            validator: Validator object to validate configurations runs
+            output_path: Output directory of the Configurator.
             executable_path: Executable of the configurator for Sparkle to call
             settings_path: Path to the settings file for the configurator
-            result_path: Path for the result files of the configurator
             configurator_target: The wrapper algorithm to standardize configurator
                 input/output towards solver wrappers.
             tmp_path: Path for the temporary files of the configurator, optional
             multi_objective_support: Whether the configurator supports
                 multi objective optimization for solvers.
         """
-        self.configurator_path = configurator_path
+        self.validator = validator
+        self.output_path = output_path
         self.executable_path = executable_path
         self.settings_path = settings_path
-        self.result_path = result_path
         self.configurator_target = configurator_target
         self.tmp_path = tmp_path
         self.multiobjective = multi_objective_support
-
-        self.scenarios_path = self.configurator_path / "scenarios"
-        self.instances_path = self.scenarios_path / "instances"
-
-        if not self.configurator_path.is_dir():
-            print(f"The given configurator path '{self.configurator_path}' is not a "
-                  "valid directory. Abort")
-            sys.exit(-1)
-
         self.scenario = None
-        self.sbatch_filename = ""
-        (self.configurator_path / "tmp").mkdir(exist_ok=True)
 
-        self.objectives = sgh.settings.get_general_sparkle_objectives()
+        self.objectives = gv.settings.get_general_sparkle_objectives()
         if len(self.objectives) > 1 and not self.multiobjective:
             print("Warning: Multiple objectives specified but current configurator "
                   f"{self.configurator_path.name} only supports single objective. "
                   f"Defaulted to first specified objective: {self.objectives[0].name}")
 
     def configure(self: Configurator,
```

Comment:
To all abstract methods (those who get functionality by the implementation), please add ` @abstractmethod` which you can get from: `from abc import abstractmethod`